engine/src/main/java/nl/inl/blacklab/search/BlackLabIndexAbstract.java

```diff
@@ -73,13 +75,18 @@ public abstract class BlackLabIndexAbstract implements BlackLabIndexWriter {
 
     protected static final Logger logger = LogManager.getLogger(BlackLabIndexAbstract.class);
 
-
     // Instance variables
     //---------------------------------------------------------------
 
     /** BlackLab instance used to create us */
     private final BlackLabEngine blackLab;
 
+    /** Are we running from inside solr? */
+    private boolean runningFromSolr = false; // TODO: pass this in through the constructor
```

Comment:
Why not pass in the BLIndexObjectFactory directly?

Reply:
Good one. I've opted to instead let BlackLabEngine "own" the BLIndexObjectFactory so no extra constructor parameter is needed.

We do still need a way to decide whether we're running from Solr. I guess we should simply pass that as a parameter when creating BlackLabEngine (or calling BlackLab.implicitEngine()).

Reply:
I've just added BlackLabEngine.setIndexObjectFactory() now to switch to Solr-based indexing, which is essentially what you were originally suggesting I think. The default is indexing directly to Lucene, of course.